src/System.IO.Pipelines.Networking.Tls/OpenSslSecurityContext.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.IO.Pipelines.Networking.Tls.Internal.OpenSsl;
+using System.Runtime.InteropServices;
+
+namespace System.IO.Pipelines.Networking.Tls
+{
+    public class OpenSslSecurityContext : IDisposable
+    {
+        //Current fixed block size (4k - 1 64 byte cacheline, should be from the pipeline factory in the future
+        internal const int BlockSize = 1024 * 4 - 64; 
+
+        private readonly string _hostName;
+        private readonly PipelineFactory _pipelineFactory;
+        private readonly bool _isServer;
+        private InteropKeys.PK12Certifcate _certifcateInformation;
+        private byte[] _alpnSupportedProtocolsBuffer;
+        private GCHandle _alpnHandle;
+        internal IntPtr SslContext;
+        internal Interop.alpn_cb AlpnCallback;
+        private ApplicationProtocols.ProtocolIds _alpnSupportedProtocols;
+
+        public OpenSslSecurityContext(PipelineFactory pipelineFactory, string hostName, bool isServer, string pathToPfxFile, string password)
+            : this(pipelineFactory, hostName, isServer, pathToPfxFile, password, 0)
+        {
+        }
+
+        public OpenSslSecurityContext(PipelineFactory pipelineFactory, string hostName, bool isServer, string pathToPfxFile, string password, ApplicationProtocols.ProtocolIds alpnSupportedProtocols)
```

Comment:
Well yeah it's not supported on a bunch of OpenSSL as well (pre 1.02) I would rather revisit than only block for 1 of potentially a number of unsupported scenarios